Fix text visibility in select inputs

ralucaStan opened the pull request:
Fixes WOOPLUG-5958

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This PR fixes the text visibility in select inputs inside the Checkout block.
We're using the same font styles as the TextInput. 

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes # .

(For Bug Fixes) Bug introduced in PR #59787, coming in v10.4

### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| Before | After |
| ------ | ----- |
|     <img width="826" height="111" alt="Screenshot 2025-12-04 at 17 34 18" src="https://github.com/user-attachments/assets/971cfa1a-35a5-4978-9a66-a74174cf1e2b" /> |    <img width="823" height="108" alt="Screenshot 2025-12-04 at 17 34 36" src="https://github.com/user-attachments/assets/735c98ea-0d9d-4d5a-895c-c65dcd067048" /> |


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add a product to cart and visit the checkout
2. Select Uruguay as a Shipping country
3. Notice the full text is visible (the tail of the g is also visible)

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Milestone

<!-- DO NOT remove or modify this section (other than to check the box). -->

<!-- milestone-target-selection -->
- [ ] Automatically assign milestone for the **[next WooCommerce version](../blob/trunk/plugins/woocommerce/woocommerce.php#L6)**
<!-- /milestone-target-selection -->

> **Note:** Check the box above to have the milestone automatically assigned when merged.
> Alternatively (e.g. for point releases), manually assign the appropriate milestone.


### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

ralucaStan committed:
Fix glyph’s descenders visibility in select inputs

ralucaStan committed:
changelog

ralucaStan commented on plugins/woocommerce/client/blocks/assets/js/base/components/select/style.scss:
```diff
@@ -29,8 +29,7 @@
 		@include reset-typography();
-		font-size: 16px;
-		line-height: 22px;
+		@include font-regular-locked();
```
We're using the same font styles as the TextInput.

ralucaStan committed:
Fix mixin

ralucaStan commented on plugins/woocommerce/client/blocks/assets/js/base/components/select/style.scss:
```diff
@@ -29,8 +29,7 @@
 		@include reset-typography();
-		font-size: 16px;
-		line-height: 22px;
+		@include font-regular-locked();
```
@alefesouza @straku could you help out with fixing this?

> The hardcoded height calculation at style.scss:49 (calc(15px + 16px * 2 + 1.5px * 2)) appears incorrect - the comment says it's based on "font line-height + vertical padding + focused top/bottom border" but uses 15px while the new line-height is 20px. This mismatch may cause layout issues.

The calculation was once modified to 20px line height, https://github.com/woocommerce/woocommerce/pull/59754/files#diff-6436e2fe82584f233917296242273149041de6adc72c66ad3d38248dd0d746c9R50.

But I also see this version https://github.com/woocommerce/woocommerce/pull/59787/files#diff-6436e2fe82584f233917296242273149041de6adc72c66ad3d38248dd0d746c9R50

If you look on trunk the Text Input uses `@include font-regular-locked`, but the Select diverged to
`font-size: 16px;
line-height: 22px;
`.

straku replied:
The source of this divergence from my initial version is these two commits:
- [Update fields styling](https://github.com/woocommerce/woocommerce/pull/59787/commits/a1ea228b196e521ee7bb13123a408e69a76536cc)
- [Fix text input font size](https://github.com/woocommerce/woocommerce/pull/59787/commits/8126e947115916e249c5db24c341e65a317a64b8)

Both come from @alefesouza, so I will have to divert this question to him as I don't know the reasoning behind them 🙇

alefesouza replied:
I made these changes to make the select field have a 50px height as seen on the designs JDhjl9Hr22AJIduP8gZ8kI-fi-5177_35349

alefesouza replied:
Also the changes in the line-height was made to make it more pixel perfect with the designs, with the `line-height: 22px` the text is a little below, perfectly matching the designs.

https://github.com/user-attachments/assets/5aa3d14e-7ae1-4c8b-8eea-bbc8b049fc46

ralucaStan replied:
Thank you for sharing the design. Although [Fix text input font size](https://github.com/woocommerce/woocommerce/pull/59787/commits/8126e947115916e249c5db24c341e65a317a64b8) changes both inputs, on trunk only the Select has the new font size, so there must have been an issue when merging.

The change didn't update the height calculations, which mentioned the old line-height. 

And it also didn't allow for enough space for letters like `g` to be fully visible. 

How do you recommend we should fix this?

ralucaStan replied:
I tried a new approach in 69a7b4c61915df649a10ed4ffebc57b779b34346. 

@alefesouza let me know what you think

alefesouza replied:
On Tsubaki theme, I'm getting padding issues on the select with this approach, it's not aligned with the label.

<img width="528" height="387" alt="Screenshot 2025-12-05 at 9 58 04 AM" src="https://github.com/user-attachments/assets/7460db6e-b309-480c-aea7-3634cf573cd0" />

alefesouza replied:
Although it's perfect with `padding: 16px 9px 0;`

alefesouza replied:
When focused, the select value goes right a little bit, so it's good to add `padding: 16px 8.5px 0;` to the `:focus` to fix it.

https://github.com/user-attachments/assets/81fa4a2d-61df-4d8a-88b4-bef9a6013fdd

ralucaStan replied:
Would you like to add those changes?

alefesouza replied:
Done on https://github.com/woocommerce/woocommerce/pull/62268/commits/378aaf4ed33ed2186eec5dcce28cfe017bac90bc, tested on Tsubaki and Storefront.

ralucaStan committed:
Set height for input and select
Remove padding on focus select to prevent content jump
Allow for a big lineheight like in the design

alefesouza committed:
Fix select padding

ralucaStan commented on plugins/woocommerce/client/blocks/packages/components/text-input/style.scss:
```diff
@@ -42,7 +42,7 @@
 	input[type="email"] {
 		@include reset-typography();
-		@include font-regular-locked;
+		@include font-for-inputs-locked();
```
The checkout block has received a facelift, and these changes are part of a bigger change, which will be announced in the release notes.

ralucaStan reviewed (commented):
I've tested on 4 different browsers, with Storefront, TT5 and Tsubaki and on mobile. Works as expected.

@alefesouza could you do a code review and approve the PR? As an author I can't approve it :)

alefesouza reviewed (approved):
Works as expected, the text visibility issues on select inputs are fixed.

alefesouza committed:
Merge branch 'trunk' into fix/select-typography
